### PR TITLE
ledger-tool: Notify geyser of transactions

### PR DESCRIPTION
**Pull request by bennofs**
#### Problem

Geyser plugins are not notified of transactions during ledger replay when using ledger tool.

#### Summary of Changes

Notify geyser plugins of transactions during ledger tool replay.

### CODE REVIEW

File: ledger-tool/src/main.rs [hunk 1]
```diff
@@ -1238,19 +1241,38 @@ fn load_bank_forks(
...
+    let (transaction_status_sender, transaction_status_service) = if transaction_notifier.is_some() {
+        let (transaction_status_sender, transaction_status_receiver) = unbounded();
+        let transaction_status_service = TransactionStatusService::new(
+            transaction_status_receiver,
+            Arc::default(),
+            false,
+            transaction_notifier,
+            blockstore.clone(),
+            false,
+            &exit.clone(),
```
Comment (lijunwangs) [inline ledger-tool/src/main.rs:1253]: remove unnecessary clone -- it is cloned inside the callee. CI also complains about it.

**Comment by lijunwangs** [comment]
Thanks for the PR. Can you document on the tests you have done and show it is working? @bennofs

**Comment by bennofs** [comment]
> Thanks for the PR. Can you document on the tests you have done and show it is working? @bennofs

I wrote a simple geyser plugin that just logs all transactions:

```rust
use env_logger::Env;
use solana_geyser_plugin_interface::geyser_plugin_interface::GeyserPlugin;
use log::info;

#[derive(Debug)]
struct LogPlugin;

impl GeyserPlugin for LogPlugin {
    fn name(&self) -> &'static str {
        "logger-plugin"
    }

    fn on_load(&mut self, _config_file: &str) -> solana_geyser_plugin_interface::geyser_plugin_interface::Result<()> {
        info!("example plugin loaded");
        Ok(())
    }

    fn notify_transaction(
        &mut self,
        transaction: solana_geyser_plugin_interface::geyser_plugin_interface::ReplicaTransactionInfoVersions,
        slot: u64,
    ) -> solana_geyser_plugin_interface::geyser_plugin_interface::Result<()> {
        match transaction {
            solana_geyser_plugin_interface::geyser_plugin_interface::ReplicaTransactionInfoVersions::V0_0_1(v1) => info!("{} transaction v1 {:?}", slot, v1),
            solana_geyser_plugin_interface::geyser_plugin_interface::ReplicaTransactionInfoVersions::V0_0_2(v2) => info!("{} transaction v2 {:?}", slot, v2),
        }
        Ok(())
    }
    
    fn transaction_notifications_enabled(&self) -> bool {
        // we cannot know yet whether the plugin needs this
        // so be cautiuous and enable always
        true
    }
}

#[no_mangle]
#[allow(improper_ctypes_definitions)]
/// # Safety
///
/// This function returns the GeyserPluginPostgres pointer as trait GeyserPlugin.
pub unsafe extern "C" fn _create_plugin() -> *mut dyn GeyserPlugin {
    env_logger::Builder::from_env(Env::new().default_filter_or("info")).init();
    let plugin = Box::new(LogPlugin);
    info!("log plugin initialized");
    Box::into_raw(plugin)
}
```

Then I checked for a ledger produced by `solana-test-validator` that `solana-ledger-tool replay` with the plugin loaded that the reported transactions look good and that the count matches the number of transactions in the ledger.

**Review by lijunwangs — APPROVED** [review]
Looks good

**Pull request merged**